crest/Assets/Crest/Crest/Scripts/Shapes/FFT/ShapeFFT.cs

```diff
@@ -432,20 +456,32 @@ void InitBatches()
                     {
                         _matGenerateWavesGeometry = new Material(Shader.Find("Crest/Inputs/Animated Waves/Gerstner Geometry"));
                     }
+
+                    Helpers.SetBlendFromPreset(_matGenerateWavesGeometry, _blendMode);
                 }
             }
 
             var usingGeometryToGenerate = _inputMode == Mode.Spline;
             if (!usingGeometryToGenerate || MeshForGeneration)
             {
+                // Queue determines draw order of this input. GetSiblingIndex gives intuitive ordering.
+                var queue = _queue + transform.GetSiblingIndex();
```

Comment:
I think this should work well. LMKWYT

Reply:
I think there are potential difficulties. I could set the queue to -1 to make it earlier, but if it's the 5th sibling it will end up in queue 4, even if other siblings are not wave inputs. And this could change of I reparent the GO.

It's a bit of a flawed approach generally but the idea of having it 10+index is the 10 would push it to the end of the"normal" range, then adding index should not reorganise within that range. Even that is a bit hopeful and has fail cases too. 

Oh maybe I have an idea, will propose via code

Reply:
Ok its trivial so just writing here:

`var queue = 1000 * _queue + transform.GetSiblingIndex();`

If the sibling index is less than 1000, this will successfully sort on *both* `_queue` and sibling index.

The remaining question mark is if i have other animated waves inputs, like register height input which should also take a factor of 1000 to sort consistently against ShapeFFT

```
        protected virtual bool GetQueue(out int queue)
        {
            var rend = GetComponent<Renderer>();
            if (rend && rend.sharedMaterial != null)
            {
                queue = 1000 * rend.sharedMaterial.renderQueue;
                return true;
            }
            queue = int.MinValue;
            return false;
        }
```

I sense that maybe this is getting thorny in terms of understanding and maintenance. Maybe changing CrestSortedList.Add to take a subindex forces the issue to all the call sites, and allows calling code to give a sibling index or other subindex or simply provide 0 if they dont care (additive blended things won't):

```
        public void Add(TKey key, TKey subKey, TValue value)
        {
            _backingList.Add(new KeyValuePair<TKey, TValue>(key * 1000 + subKey, value));
            _needsSorting = true;
        }
```

I haven't followed this through but could try it out if you agree to see if it works.

Reply:
Good idea. I don't think it makes sense to change the sorted list. This is specific to inputs at the moment so could implement logic in base class? Queue would be a good feature across the board. A separate PR is warranted. Happy to do it tomorrow if you're busy today.

Reply:
I think a static method on register input base class could work.